src/core/ir/eval.rs

```diff
@@ -68,20 +68,42 @@ impl IR {
                         eval_io(io, ctx).await
                     }
                 }
-                IR::Map(Map { input, map }) => {
-                    let value = input.eval(ctx).await?;
-                    if let ConstValue::String(key) = value {
+                IR::Map(Map { input, map }) => match input.eval(ctx).await? {
+                    ConstValue::Null => Ok(ConstValue::Null),
+                    ConstValue::String(key) => {
                         if let Some(value) = map.get(&key) {
                             Ok(ConstValue::String(value.to_owned()))
                         } else {
                             Err(Error::ExprEval(format!("Can't find mapped key: {}.", key)))
                         }
-                    } else {
-                        Err(Error::ExprEval(
-                            "Mapped key must be string value.".to_owned(),
-                        ))
                     }
-                }
+                    ConstValue::List(vec) => {
+                        let vec = vec
+                            .into_iter()
+                            .map(|value| {
+                                if let ConstValue::String(key) = value {
+                                    if let Some(value) = map.get(&key) {
+                                        Ok(ConstValue::String(value.to_owned()))
+                                    } else {
+                                        Err(Error::ExprEval(format!(
+                                            "Can't find mapped key: {}.",
+                                            key
+                                        )))
+                                    }
+                                } else {
+                                    Err(Error::ExprEval(
+                                        "Mapped key must be either string or array value."
+                                            .to_owned(),
```

Comment:
better move this to separate function that can be called recursively. It'll lower code duplication and will be able to handle lists inside lists